packages/javascript-sdk/src/token-manager/helpers.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * @forgerock/javascript-sdk
+ *
+ * helpers.ts
+ *
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+/**
+ * @module
+ * @ignore
+ * These are private utility functions for Token Manager
+ */
+
+export function tokensWillExpireWithinThreshold(
+  oauthThreshold: number,
+  tokenExpiry: number,
+  timeNow: number,
+): boolean {
+  return tokenExpiry - oauthThreshold < timeNow;
```

Comment:
I believe we can reduce the API of this function since `timeNow` is always `Date.now`.  If there is a case where `timeNow` can be something other than `Date.now` perhaps we can assign it a default value of `Date.now` so we can omit this argument from the calling code unless explicitly required.

```suggestion
export function tokensWillExpireWithinThreshold(
  oauthThreshold: number,
  tokenExpiry: number,
): boolean {
  const timeNow = Date.now();
  return tokenExpiry - oauthThreshold < timeNow;
```

Reply:
Ok I see your point. I was thinking from a testing point of view it would be better to pass in the date for predictability - that was the main reason for passing it in. If we were to go with your suggested change, how might we test this without having Date.now() in our tests?

Reply:
I think you have to mock the the Date object. Something like

```
describe('Test', () => {
  // to avoid race conditions between expected and actual date values
  beforeAll(() => {
    jest.useFakeTimers('modern'); // tell Jest to use a different timer implementation.
    jest.setSystemTime(new Date('20 Aug 2020 00:12:00 GMT').getTime())
  });

  afterAll(() => {
    // Back to reality...
    jest.useRealTimers();
  });
  ```
  https://github.com/facebook/jest/issues/2234#issuecomment-731451275

Reply:
Makes sense, I'll have a go at mocking the date. Thanks

Reply:
This worked easily enough - thanks 👍